pkg/jx/cmd/common_install.go

```diff
@@ -1069,65 +1172,11 @@ func (o *CommonOptions) installAws() error {
 	return nil
 }
 
-func (o *CommonOptions) installEksCtl() error {
-	binDir, err := util.JXBinLocation()
-	binary := "eksctl"
-	if err != nil {
-		return err
-	}
-	fileName, flag, err := o.shouldInstallBinary(binDir, binary)
-	if err != nil || !flag {
-		return err
-	}
-	latestVersion, err := util.GetLatestVersionStringFromGitHub("weaveworks", binary)
-	if err != nil {
-		return err
-	}
-	extension := "tar.gz"
-	if runtime.GOOS == "windows" {
-		extension = "zip"
-	}
-	clientURL := fmt.Sprintf("https://github.com/weaveworks/eksctl/releases/download/%s/eksctl_%s_%s.%s", latestVersion, strings.Title(runtime.GOOS), runtime.GOARCH, extension)
-	fullPath := filepath.Join(binDir, fileName)
-	tarFile := fullPath + "." + extension
-	err = o.downloadFile(clientURL, tarFile)
-	if err != nil {
-		return err
-	}
-	if extension == "zip" {
-		zipDir := filepath.Join(binDir, "eksctl-tmp-"+uuid.NewUUID().String())
-		err = os.MkdirAll(zipDir, DefaultWritePermissions)
-		if err != nil {
-			return err
-		}
-		err = util.Unzip(tarFile, zipDir)
-		if err != nil {
-			return err
-		}
-		f := filepath.Join(zipDir, fileName)
-		exists, err := util.FileExists(f)
-		if err != nil {
-			return err
-		}
-		if !exists {
-			return fmt.Errorf("Could not find file %s inside the downloaded eksctl.zip!", f)
-		}
-		err = os.Rename(f, fullPath)
-		if err != nil {
-			return err
-		}
-		err = os.RemoveAll(zipDir)
-	} else {
-		err = util.UnTargz(tarFile, binDir, []string{binary, fileName})
-	}
-	if err != nil {
-		return err
-	}
-	err = os.Remove(tarFile)
-	if err != nil {
-		return err
-	}
-	return os.Chmod(fullPath, 0755)
+func (o *CommonOptions) installEksCtl(skipPathScan bool) error {
+	return o.installOrUpdateBinary("eksctl",
+		 "weaveworks",
+		"https://github.com/weaveworks/eksctl/releases/download/{{.version}}/eksctl_{{.os}}_{{.arch}}.{{.extension}}",
+		"0.1.1", skipPathScan)
```

Comment:
I think, you should pass in an empty version to really get the latest version from GitHub. Otherwise, it will always download `0.1.1`.

Reply:
Actually this is for purpose. Given version of jx should use the fixed version of binary dependency, for example eksctl. This way you can always be sure for example `jx 0.15 => eksctl 0.1` and `jx 0.16 => eksctl 0.3`. And you know that jx has downloaded version of binary dependency which has been used and tested by jx team. jx should control which version of dependency it is downloading, the same way you specify exact version of jar dependencies in Maven project POM instead of using the latest one.

Reply:
Got it. At least would be nice to keep the versions of all binaries in a central place, instead of hard-code it into the `installEksCtrl` method. We can have something like: `installEksCtrl(version string, skipPathScan bool). For instance, in this case you can provide the fixed version from upgrade binaries command.